src/main.rs

```diff
@@ -583,18 +583,20 @@ async fn start_ldk() {
 	// to avoid churn in the global network graph.
 	let chan_manager = Arc::clone(&channel_manager);
 	let network = args.network;
-	tokio::spawn(async move {
-		let mut interval = tokio::time::interval(Duration::from_secs(60));
-		loop {
-			interval.tick().await;
-			for chan_info in chan_manager.list_usable_channels() {
-				if chan_info.is_public && args.ldk_announced_listen_addr.is_some() {
-					chan_manager.broadcast_node_announcement([0; 3], args.ldk_announced_node_name,
-						vec![args.ldk_announced_listen_addr.as_ref().unwrap().clone()]);
+	if args.ldk_announced_listen_addr.is_some() {
+		tokio::spawn(async move {
+			let mut interval = tokio::time::interval(Duration::from_secs(60));
+			loop {
+				interval.tick().await;
+				for chan_info in chan_manager.list_usable_channels() {
+					if chan_info.is_public {
+						chan_manager.broadcast_node_announcement([0; 3], args.ldk_announced_node_name,
+							vec![args.ldk_announced_listen_addr.as_ref().unwrap().clone()]);
+					}
 				}
```

Comment:
Should this only broadcast once rather than for each public channel?

```rust
let has_public_channels = chan_manager
	.list_usable_channels()
	.iter()
	.filter(|chan_info| chan_info.is_public)
	.count() > 0;
```